goal_src/engine/ui/progress-h.gc

```diff
@@ -128,6 +130,16 @@
   :flag-assert         #x900000028
   )
 
+;; new custom type to extend the very limited progress-menu system
+;; the user can only interact with one element at a time, so a single global is sufficient
+(deftype progress-menu-list-tracker (basic)
+  ((direction symbol :offset-assert 4) ; 'left | 'right
+   (transition? symbol :offset-assert 8) ; '??
+   (x-offset int32 :offset-assert 12)
+   (selected-index int32 :offset-assert 16)))
+
+;; why a new global? because i dont want to modify the type below....but that would get me what i want as well!
+(define *progress-menu-list-tracker* (the-as progress-menu-list-tracker #f))
```

Comment:
where do you actually allocate this?
My preference would be to put `(define *progress-menu-list-traker* (new 'static 'progress-menu-list-tracker))` here.